### PR TITLE
reduce_logsumexp fix for reduction_indices.  Fixes issue #5291

### LINKED ISSUE #5291: [Bug] reduce_logsumexp examples do not work in 0.11.0rc1
Version: 0.11.0rc1

These examples do not work:

```
# 'x' is [[0, 0, 0]]
#         [0, 0, 0]]
tf.reduce_logsumexp(x, 0) ==> [log(2), log(2), log(2)]
tf.reduce_logsumexp(x, 1) ==> [log(3), log(3)]
```

Because:
`tf.squeeze(t, squeeze_dims)` does not take integer for `squeeze_dims`

### PR DESCRIPTION
**Pull request by hoangmit**
Fix #5291

### CODE REVIEW

**Comment by hoangmit** [comment]
I signed it!

File: tensorflow/python/ops/math_ops.py [hunk 1]
```diff
@@ -1389,6 +1389,7 @@ def reduce_logsumexp(input_tensor, reduction_indices=None, keep_dims=False,
         reduction_indices,
         keep_dims=True)) + my_max
     if not keep_dims:
+      reduction_indices = [reduction_indices] if isinstance(reduction_indices, int) else reduction_indices
```
Comment (vrv) [inline tensorflow/python/ops/math_ops.py]: Change this to:

```
reduction_indices = _ReductionDims(input_tensor, reduction_indices)
```

That's what we do for the other reduction functions in this file.

File: tensorflow/python/ops/math_ops.py [hunk 2]
```diff
@@ -1389,6 +1389,7 @@ def reduce_logsumexp(input_tensor, reduction_indices=None, keep_dims=False,
         reduction_indices,
         keep_dims=True)) + my_max
     if not keep_dims:
+      reduction_indices = _ReductionDims(input_tensor, reduction_indices)
```
Comment (vrv) [inline tensorflow/python/ops/math_ops.py]: Oh, is it possible to have a test for triggering this case?

Comment (vrv) [inline tensorflow/python/ops/math_ops.py]: (See where we currently test reduce_logsumexp and add another case for this).

File: tensorflow/python/ops/math_ops_test.py [hunk 3]
```diff
@@ -68,6 +68,17 @@ def testReductionIndices(self):
         y_tf_np = y_tf.eval()
         self.assertAllClose(y_tf_np, y_np)
 
+
```
Comment (vrv) [inline tensorflow/python/ops/math_ops_test.py]: only one line space between tests

**Comment by vrv** [comment]
@tensorflow-jenkins test this please

Nm, I did it myself.

**Comment by vrv** [comment]
Well that's unfortunate, it looks like the argument to squeeze has to be an array, not a Tensor, which means we can't use _ReductionDims().

**Comment by hoangmit** [comment]
Use earlier change instead

**Comment by vrv** [comment]
@tensorflow-jenkins test this please

**Pull request merged**